### PR TITLE
Removed references to Setuptools/easy_install in favor of pip.  

### PR DESCRIPTION
**Pull request by adamn**
A new Python user hardly needs to know that easy_install ever existed.  

Also addresses #676 for OS X.  Open to suggestions.

### CODE REVIEW

**Comment by kennethreitz** [comment]
Setuptools is still required for Pip to operate. 

I don't like how the virtualenv section was converted to Python3.

File: docs/dev/virtualenvs.rst [hunk 1]
```diff
@@ -8,15 +8,22 @@ projects in separate places, by creating virtual Python environments for them.
 It solves the "Project X depends on version 1.x but, Project Y needs 4.x"
 dilemma, and keeps your global site-packages directory clean and manageable.
 
-For example, you can work on a project which requires Django 1.3 while also
-maintaining a project which requires Django 1.0.
+For example, you can work on a project which requires Django 1.9 while also
+maintaining a project which requires Django 1.8.
+
+If you are using Python 3, make sure you run something like the following (or
+have it in your env variables in `~/.bashrc`):
```
Comment (sigmavirus24) [inline docs/dev/virtualenvs.rst]: To be valid reStructuredText, this should have two backticks on either side, e.g.,

```
in ``~/.bashrc``):
```

Comment (adamn) [inline docs/dev/virtualenvs.rst]: fixed.

**Comment by adamn** [comment]
Setuptools is required for Pip but since users aren't directly using Setuptools in the installation section it seems ancillary.  Any Setuptools discussion might better belong in the Packaging section since beginning users only need to know about pip IMHO.

Anyway, I'm happy to undo that.

As for the virtualenv changes, if a Python 3 user reads that section without the Python 3 information, they will get very lost.  Virtualenv will appear to work but the environments will all be Python2.  We need to help out users who have installed Python3 and want virtualenv to function the way the docs say it will.

I've updated that change to be less vocal - let me know what you think.  I still think the -p flag example should simply point to a python3 version since that is the common use case.  These days more beginning people are likely to us Python 2 and 3 side by side rather than two different Python 2s.  Non-beginners will understand that -p allows for any python interpreter.

**Comment by kennethreitz** [comment]
@adamn I think a better approach for the virtualenv section is to return it to it's previous (Python 2) state, and perhaps add a section explaining the use of the `$ virtualenv -p python3` option. Most people don't know about this, and explaining it to them right away may make things much easier for them down the road (as well as help them understand what virtualenv is actually *doing*).

**Comment by kennethreitz** [comment]
seutptools still needs to be restored to `installation.rst` :)

File: docs/starting/install/osx.rst [hunk 2]
```diff
@@ -3,13 +3,16 @@
...
 
+.. note::
+    Check out our :ref:`guide for installing Python 3 on OS X<install3-osx>`.
+
 The latest version of Mac OS X, El Capitan, **comes with Python 2.7 out of the box**.
 
-You do not need to install or configure anything else to use Python. Having
-said that, I would strongly recommend that you install the tools and libraries
-described in the next section before you start building Python applications
-for real-world use. In particular, you should always install Setuptools, as it
```
Comment (kennethreitz) [inline docs/starting/install/osx.rst]: I also don't think removing setuptools here is a good idea. It's important to know about these tools, even if you aren't going to use them directly.

File: docs/starting/install/osx.rst [hunk 3]
```diff
@@ -33,11 +36,11 @@ package.
     diagnose.
 
 .. note::
-    If you perform a fresh install of XCode, you will also need to add the 
+    If you perform a fresh install of XCode, you will also need to add the
     commandline tools by running ``xcode-select --install`` on the terminal.
 
 While OS X comes with a large number of UNIX utilities, those familiar with
-Linux systems will notice one key component missing: a decent package manager.
```
Comment (kennethreitz) [inline docs/starting/install/osx.rst]: please restore 'decent' :)

**Comment by kennethreitz** [comment]
According to @dstufft, pip plans to automatically install setuptools in the future. Let's hold off on this until then.

**Comment by adamn** [comment]
I re-instated the setuptools references.  Maybe we can just try to get the Python3 Mac docs up and close this?  A new ticket can be opened if/when pip automatically installs setuptools in the future.

**Comment by kennethreitz** [comment]
+1

**Comment by kennethreitz** [comment]
This appears to require a rebase. 

-----------


One of these days, I intend to sit down and refactor these with Python 3 guides available as well. I have a feeling I'll start spending more time on this project very soon.

Hopefully I'm right :)

**Comment by adamn** [comment]
Should I rebase this or just delete it?

**Comment by kennethreitz** [comment]
i think i will merge it, with a few changes

**Comment by adamn** [comment]
I'm also happy to rebase/update, just let me know.  Otherwise, maybe we should just get this in and do other updates in another PR

**Comment by kennethreitz** [comment]
whatever lets me do the "merge" button easily :)

**Comment by kennethreitz** [comment]
"This branch has conflicts that must be resolved"

**Comment by kennethreitz** [comment]
I could always resolve myself, but.. you know :)

**Comment by kennethreitz** [comment]
are you ever going to make these changes?

**Comment by adamn** [comment]
If there's more you need, just let me know.

**Comment by kennethreitz** [comment]
i made the changes myself :)

**Comment by kennethreitz** [comment]
thank you very much! We need to update the other two installation guides now :)

**Pull request merged**